kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/CachingKms.java

```diff
@@ -89,6 +100,17 @@ public Serde<E> edekSerde() {
     @NonNull
     @Override
     public CompletionStage<K> resolveAlias(@NonNull String alias) {
-        return resolveAliasCache.get(alias);
+        Long notFoundSince = notFoundAliasCache.getIfPresent(alias);
+        if (notFoundSince != null) {
+            return CompletableFuture.failedFuture(new UnknownAliasException("alias " + alias + " not found, cached since " + notFoundSince));
+        }
+        CompletableFuture<K> resolved = resolveAliasCache.get(alias);
+        resolved.whenComplete((k, throwable) -> {
+            if (throwable instanceof UnknownAliasException || (throwable instanceof CompletionException && throwable.getCause() instanceof UnknownAliasException)) {
+                LOGGER.debug("caching unknown alias {} for {}", alias, CACHE_NOT_FOUND_ALIAS_DURATION);
+                notFoundAliasCache.put(alias, System.currentTimeMillis());
```

Comment:
For this usage I figured we do not need to introduce further co-ordination to try and prevent parallel calls to the KMS, this is a good enough measure to stop the bulk of requests to the KMS